src/Raven.Server/RavenServer.cs

```diff
@@ -379,20 +396,43 @@ public DynamicJsonValue ToJson()
                     [nameof(CurrentConsumption)] = CurrentConsumption,
                     [nameof(MachineCpuUsage)] = MachineCpuUsage,
                     [nameof(History)] = historyByMinute,
+                    [nameof(LastSyncTime)] = LastSyncTime
                 };
             }
         }
 
         private void StartMonitoringCpuCredits()
         {
             CpuCreditsBalance.Used = true;
-            CpuCreditsBalance.RemainingCpuCredits = CpuCreditsBalance.BaseCredits;
             CpuCreditsBalance.BackgroundTasksThresholdReleaseValue = CpuCreditsBalance.BackgroundTasksThreshold * 1.25;
             CpuCreditsBalance.FailoverThresholdReleaseValue = CpuCreditsBalance.FailoverThreshold * 1.25;
             CpuCreditsBalance.CreditsGainedPerSecond = CpuCreditsBalance.BaseCredits / 3600;
 
             int remainingTimeToBackgroundAlert = 15, remainingTimeToFailvoerAlert = 5;
             AlertRaised backgroundTasksAlert = null, failoverAlert = null;
+
+            if (string.IsNullOrEmpty(Configuration.Server.CpuCreditsExec) == false)
+            {
+                var response = GetCpuCreditsFromExec();
+
+                CpuCreditsBalance.RemainingCpuCredits = response.Remaining;
+                CpuCreditsBalance.LastSyncTime = response.Timestamp;
+
+                CpuCreditsBalance.History = new double[60 * 60];
+                CpuCreditsBalance.HistoryCurrentIndex = 0;
+            }
+            else
+            {
+                var print = RavenConfiguration.GetKey(x => x.Server.CpuCreditsExec) +
```

Comment:
Is it mandatory now?

Reply:
I think it should be mandatory if you supply other CPU credits configurations.

Reply:
so it should throw, not log

Reply:
yes